v2_x/git-remote-gosh/src/blockchain/tag/save.rs

```diff
@@ -61,7 +64,29 @@ impl Tagging for Everscale {
         content: String,
         commit_address: BlockchainContractAddress,
     ) -> anyhow::Result<()> {
-        let wallet_contract = wallet.take_zero_wallet().await?;
+        let client = Arc::clone(&self.ever_client);
+        let GetContractCodeResult { code } =
+            crate::blockchain::get_contract_code(
+                &client,
+                &self.repo_contract.get_address(),
+                crate::blockchain::ContractKind::Tag
+            )
+            .await?;
+
+        let address = calculate_contract_address(
+            &client,
+            crate::blockchain::ContractKind::Tag,
+            &code,
+            Some(serde_json::json!({ "_nametag": tag_name })),
+        )
+        .await?;
+
+        let expected_tag_contract = GoshContract::new(&address, gosh_abi::TAG);
+        if expected_tag_contract.is_active(&client).await? {
+            anyhow::bail!("hint: Updates were rejected because the tag already exists in the remote.");
+        }
+
+        let wallet_contract = wallet.take_one().await?;
```

Comment:
Same as in the v1, tag should be deployed via zero wallet

Reply:
Fixed